### PR TITLE
Remove LocationXY usage on ride and peep

### PR DESCRIPTION
**Pull request by tupaschoal**
Helps #10267

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -3574,7 +3574,7 @@ static void vehicle_update_collision_setup(rct_vehicle* vehicle)
 }
 
 /** rct2: 0x009A3AC4, 0x009A3AC6 */
-static constexpr const LocationXY16 stru_9A3AC4[] = {
+static constexpr const CoordsXY stru_9A3AC4[] = {
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: These are not coords. I'm not really sure what they represent yet but they aren't coords

Comment (tupaschoal) [reply src/openrct2/ride/Vehicle.cpp]: It looks like they are a collection of constant used to calculate coords for the vehicle crash

File: src/openrct2/peep/Guest.cpp [hunk 2]
```diff
@@ -1227,9 +1227,9 @@ void Guest::UpdateSitting()
         if (!(pathingResult & PATHING_DESTINATION_REACHED))
             return;
 
-        LocationXYZ16 loc = {
-            (int16_t)((x & 0xFFE0) + BenchUseOffsets[var_37 & 0x7].x),
-            (int16_t)((y & 0xFFE0) + BenchUseOffsets[var_37 & 0x7].y),
+        CoordsXYZ loc = {
+            (x & 0xFFE0) + BenchUseOffsets[var_37 & 0x7].x,
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp]: Use ToTileStart()

File: src/openrct2/peep/Guest.cpp [hunk 3]
```diff
@@ -3633,7 +3633,7 @@ static void peep_update_ride_leave_entrance_waypoints(Peep* peep, Ride* ride)
     Guard::Assert(!entranceLocation.isNull());
     uint8_t direction_entrance = entranceLocation.direction;
 
-    LocationXY16 waypoint;
+    CoordsXY waypoint;
     waypoint.x = ride->stations[peep->current_ride_station].Start.x * 32 + 16;
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp]: Use ToTileCentre()

File: src/openrct2/peep/Staff.cpp [hunk 4]
```diff
@@ -445,8 +445,7 @@ static uint8_t staff_handyman_direction_to_nearest_litter(Peep* peep)
         return 0xFF;
     }
 
-    LocationXY16 litterTile = { static_cast<int16_t>(nearestLitter->x & 0xFFE0),
-                                static_cast<int16_t>(nearestLitter->y & 0xFFE0) };
+    CoordsXY litterTile = { nearestLitter->x & 0xFFE0, nearestLitter->y & 0xFFE0 };
```
Comment (Gymnasiast) [inline src/openrct2/peep/Staff.cpp]: Use ToTileStart()

File: src/openrct2/world/Footpath.cpp [hunk 5]
```diff
@@ -70,7 +70,7 @@ const LocationXY16 BenchUseOffsets[8] = {
 };
 
 /** rct2: 0x00981D6C, 0x00981D6E */
-const LocationXY16 word_981D6C[4] = { { -1, 0 }, { 0, 1 }, { 1, 0 }, { 0, -1 } };
+const CoordsXY word_981D6C[4] = { { -1, 0 }, { 0, 1 }, { 1, 0 }, { 0, -1 } };
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp]: These look like TileCoords offsets, not Coords.

File: src/openrct2/peep/Guest.cpp (hunk 3 above)
Comment (tupaschoal) [reply src/openrct2/peep/Guest.cpp]: I've done it, but note that it only sums the 16, but doesn't `floor`s the coords first.

Comment (Gymnasiast) [reply src/openrct2/peep/Guest.cpp]: It doesn't need to - it multiplies by 32 first, so the value is a multiple of 32 already - already floored.

File: src/openrct2/ride/Vehicle.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp]: The code using them is very hard to read, but it does look like they are indeed base values for vehicle crash coords.

Not sure if we should save them in a CoordsXY value - perhaps it's better to revert this change, for now at least.

Comment (tupaschoal) [reply src/openrct2/ride/Vehicle.cpp]: Ok, since it was the only change in that commit, I dropped it altogether.

**Pull request merged**